Typescript SDK -  Get Started page

martin0995 opened the pull request:
### Description

Get Started page for the Wormhole TypeScript SDK. This page:
- Introduces the SDK’s purpose and its integration across Wormhole protocols like Token Bridge, Settlement, and CCTP
- Outlines the prerequisites and steps to initialize a new TypeScript project
- Walks through a minimal example that logs the Wormhole chain ID for Ethereum
- Includes a secondary example that inspects a deployed Token Bridge contract
- Links to next steps

This page is designed to give first-time users a fast, hands-on intro to using the Wormhole TypeScript SDK.

### Checklist

- [x] **Required** - I have added a label to this PR 🏷️
- [x] **Required** - I have run my changes through Grammarly
- [ ] If pages have been moved, I have created redirects in the `wormhole-mkdocs` repo

dawnkelly09 commented on tools/typescript-sdk/get-started.md:
````diff
@@ -0,0 +1,106 @@
+    npx tsx chain-id.ts
+    ```
+
+???- code "Expected output"
````
This should be in a terminal output element rather than a code admonition.

dawnkelly09 commented on tools/typescript-sdk/get-started.md:
````diff
@@ -0,0 +1,106 @@
+    npx tsx token-bridge.ts
+    ```
+
+???- code "Expected output"
````
Same ask as the other admonition. Please update to use terminal output element.

dawnkelly09 commented on .snippets/code/tools/typescript-sdk/get-started/snippet-2.ts:
```diff
@@ -0,0 +1,8 @@
+import { wormhole } from '@wormhole-foundation/sdk';
```
Same issue with top-level async here:
```
import { wormhole } from '@wormhole-foundation/sdk';
import evm from '@wormhole-foundation/sdk/evm';

async function main() {
    // Initialize Wormhole SDK for mainnet
    const wh = await wormhole('Mainnet', [evm]);
    // Get the Ethereum chain context
    const eth = wh.getChain('Ethereum');
    // Fetch the deployed Token Bridge contract
    const tokenBridge = await eth.getTokenBridge();
    console.log('Token Bridge:', tokenBridge);
}

main().catch((err) => {
    console.error("Error running script:", err);
    process.exit(1);
  });
```

dawnkelly09 commented on .snippets/code/tools/typescript-sdk/get-started/snippet-3.json:
```diff
@@ -0,0 +1,26 @@
+Token Bridge address: EvmTokenBridge {
```
Please update to use a terminal output element.

dawnkelly09 commented on tools/typescript-sdk/get-started.md:
````diff
@@ -0,0 +1,106 @@
+
+2. Import the SDK, initialize it in `Testnet`, retrieve the context for Ethereum, and log its Wormhole chain ID:
+
+    ```ts
````
```suggestion
    ```ts title="chain-id.ts"
```
Please check your other snippets to add the file name so it shows at the top for the user

dawnkelly09 reviewed (changes requested):
Needs a few tweaks. Thank you!

dawnkelly09 commented on .snippets/code/tools/typescript-sdk/get-started/snippet-1.ts:
```diff
@@ -0,0 +1,7 @@
+import { wormhole } from '@wormhole-foundation/sdk';
```
This wouldn't run because of a top-level async but no ts config so it didnt' know how to handle it. I wrapped it in an async function and it seems to work now. Please test on your side to be sure. thanks! 
```
import { wormhole } from '@wormhole-foundation/sdk';
import evm from '@wormhole-foundation/sdk/evm';

async function main(){
    // initialize Wormhole SDK for testnet
    const wh = await wormhole('Testnet', [evm]);
    // Get chain context for Ethereum
    const chain = wh.getChain('Ethereum');
    // Print Wormhole chain ID for Ethereum to terminal
    console.log('Wormhole chain ID for Ethereum:', chain.config.chainId);
}

main().catch((err) => {
    console.error("Error running script:", err);
    process.exit(1);
  });
  ```

Telucero commented:
Aloha, 
I ran into the same issues Dawn mentioned when testing it out, but the fixes seemed to resolve those issues. I do have a concern, when I ran the `npm install @wormhole-foundation/sdk`  I got this:

npm install @wormhole-foundation/sdk

up to date, audited 388 packages in 10s

52 packages are looking for funding
  run `npm fund` for details

**9 high severity vulnerabilities

Not sure if this is normal or not, so it could be nothing, but it may be worthwhile to make a note of this since it could worry some newer folks if they try to follow along.

eshaben commented on tools/typescript-sdk/get-started.md:
```diff
@@ -0,0 +1,106 @@
+
+## Try the SDK
+
+Explore two minimal examples to validate your setup and familiarize yourself with the SDK: log a chain ID and inspect a deployed protocol contract.
```
I'm not sold on these examples. If I was on this page, I'd want to see ok how do I initialize the SDK and how can I send a basic transfer? Validating the setup can be useful, but it's not really what people care about

eshaben commented on tools/typescript-sdk/get-started.md:
```diff
@@ -0,0 +1,106 @@
+
+Now that you’ve successfully used the SDK to interact with a supported chain and one of Wormhole’s core contracts, you’re ready to build more advanced functionality.
+
+ - [Understand Wormhole Chains](){target=\_blank} – view the list of supported networks and their Wormhole chain IDs
+ - [Use the SDK for Transfers](){target=\_blank} – move tokens cross-chain using the Token Bridge protocol
+ - [SDK Reference](){target=\_blank} – dive into the full API surface of the Wormhole SDK
```
TODOs please for the links that needs to be circled back to

dawnkelly09 commented:
> We need to come up with a standard for the tools pages. Like in a get started guide for any type of tool I would expect to see Installation, initialization, basic examples, the most important configurations. That type of thing.
> 
> I also think we should revisit the examples. What are the most common use cases for the SDK? Which features are developers most interested in? It might help to align the examples with those priorities so developers can more easily see how the SDK fits their needs. Let's reel 'em in! 🎣

I made some changes to this:

1. User accomplishes: 
- making sure they have SDK prereqs installed
- installing the SDK and TSX helper
- (optional) create a `tsconfig.json`
- Create and update `src/main.ts` to initialize the WH instance, designate "Testnet", add evm and solana platform support, and retrieve the Solana and Sepolia chain IDs and RPCs to output to the terminal to confirm success

Why? Most examples would make this too long and/or complex. This is a good base to set the user up with that we can use as a Prerequisite in guides/tutorials to ensure a shared, understood starting point. 

2. Next Steps:

This is a little beefy but if you look at this page as "Get the minimum set up and then go do next thing" I think it works. The "SDK Helpers" I need to write up but they're already outlined and won't take long. 

I tested this code and everything worked as expected with my local setup.

eshaben reviewed (commented):
This is very like start from scratch friendly, which is great, but we also need to consider that there might be devs with existing projects that are just trying to plug this into what they already have. So we should try to consider that in the layout. For example, with the suggestion I left on separating the initialization out from the examples.

I feel like we need to give users something to sink their teeth in. The examples that are here are really not providing much value. I'd much rather see a transfer than logging chain details 🤷‍♀️ . I don't think we need to walk users through every step of the transfer, just showing them how easy it is would be a win IMO

dawnkelly09 commented on tools/typescript-sdk/get-started.md:
```diff
@@ -0,0 +1,115 @@
+
+These product-focused guides offer the TL;DR on using Wormhole products and include practical instructions and example code to get your integration off to the best start: 
+<!--TODO: Do we want them all here or just some right now? Update with final list, verify titles, add links and descriptions-->
+- [Get Started with Messaging](/docs/products/messaging/get-started/){target=\_blank} - follow this guide to deploy Wormhole-based message sender and receiver smart contracts and use them to send messages across chains
```
Per our Slack convo, no. It's the Solidity SDK. I've removed and made a note to add it to Next Steps for that page when we get there.

martin0995 commented:
> Aloha, I ran into the same issues Dawn mentioned when testing it out, but the fixes seemed to resolve those issues. I do have a concern, when I ran the `npm install @wormhole-foundation/sdk` I got this:
> 
> npm install @wormhole-foundation/sdk
> 
> up to date, audited 388 packages in 10s
> 
> 52 packages are looking for funding run `npm fund` for details
> 
> **9 high severity vulnerabilities
> 
> Not sure if this is normal or not, so it could be nothing, but it may be worthwhile to make a note of this since it could worry some newer folks if they try to follow along.

Yes, that's normal and sadly out of our scope. It's part of the package itself

dawnkelly09 commented:
> I get your point, but it is mainly used for Token Bridge and CCTP. Those products have their own get started pages where we guide users through creating a transaction using the SDK. So, I think it would be too repetitive to have the same thing here. We could definitely reference those get started pages at the beginning in case the user wants to do those kinds of things with the SDK.
> 
> What do you think? @eshaben @dawnkelly09

I've updated the conclusion and Next Steps sections to build the developer journey path out a bit more and differentiate between "I want to understand the messaging basics better" and "take me to working with tokens right now" which I think is part of what we were missing. Let me know what you guys think.

martin0995 commented:
> > I get your point, but it is mainly used for Token Bridge and CCTP. Those products have their own get started pages where we guide users through creating a transaction using the SDK. So, I think it would be too repetitive to have the same thing here. We could definitely reference those get started pages at the beginning in case the user wants to do those kinds of things with the SDK.
> > What do you think? @eshaben @dawnkelly09
> 
> I've updated the conclusion and Next Steps sections to build the developer journey path out a bit more and differentiate between "I want to understand the messaging basics better" and "take me to working with tokens right now" which I think is part of what we were missing. Let me know what you guys think.

I like that! I added a mention in the introduction

eshaben commented on tools/typescript-sdk/get-started.md:
```diff
@@ -0,0 +1,126 @@
+    
+Congratulations! You’ve successfully installed the Wormhole TypeScript SDK and initialized a `Wormhole` instance. Consider the following options to build on what you've accomplished.
+
+## Next Steps
```
I'm not sold on having sections within the next steps. I think this part just needs to be as concise as possible and straight to the point.

- [Send a multichain message]()
- [Transfer wrapped assets via the Token Bridge]()
- [Transfer USDC via the CCTP Bridge]()


Not sure that NTT belongs on this list with the content that we have right now. I feel like if users are also on the SDK page, they don't want to use the Solidity SDK. So I'm not sold on pushing them there

ilariae commented on tools/typescript-sdk/get-started.md:
```diff
@@ -0,0 +1,177 @@
+
+## Introduction
+
+The Wormhole TypeScript SDK provides a unified, type-safe interface for building cross-chain applications. It is a foundational toolkit that supports interaction with core Wormhole protocols, including Token Bridge, CCTP, and Settlement, giving developers a consistent API across multiple chains.
```
```suggestion
The Wormhole TypeScript SDK provides a unified, type-safe interface for building cross-chain applications. It is a foundational toolkit that supports interaction with core Wormhole protocols, including Native Token Transfers, Token Bridge, CCTP, and Settlement, giving developers a consistent API across multiple chains.
```

ilariae commented on tools/typescript-sdk/get-started.md:
```diff
@@ -0,0 +1,177 @@
+
+## Next Steps
+
+- [Get familiar with the SDK](/tools/typescript-sdk/sdk-reference/)
```
404 link

eshaben replied:
```suggestion
- [Get familiar with the SDK](/docs/tools/typescript-sdk/sdk-reference/)
```

ilariae reviewed (commented):
looks good just a couple minor comments

dawnkelly09 reviewed (commented):
This is looking great! I updated the "note" admonitions to use "example" for now because they are purple and the little beaker looks cool with the styling. We can update styling for admos later but I like this for the moment. :-)

eshaben commented on tools/typescript-sdk/get-started.md:
````diff
@@ -79,7 +79,7 @@ You can choose to install a specific set of packages as needed. For example, to
 ```
 
-??? note "Complete list of individually published packages"
+??? example "Complete list of individually published packages"
````
This isn't really an example though

eshaben replied:
ah true, ok

ilariae reviewed (approved):
tested and lgtm!
